### PR TITLE
Fix results stacking of inline-results for normal eval

### PR DESCRIPTION
**Pull request by rundis**
Ensure inline results are cleared when evaling lines where col for result is offset from previous eval result Addressing - #2004

Seems to still work as expected with instarepl and watches as well.

### CODE REVIEW

**Comment by kenny-evitt** [comment]
This seems good. It worked for Clojure, ClojureScript, and JS files, and a Clojure InstaREPL.

**Comment by kenny-evitt** [comment]
I say merge away.

But do we need to, or should we, re-QA again?

**Comment by cldwalker** [comment]
I don't see how it's working for certain instarepl cases if https://github.com/LightTable/Clojure/commit/301729825c4e0117c59d719606142cfd603834c3 is still in Clojure. I'd rather not do this until we've understood what was fixed in instarepl and keep the instarepl fix in while fixing our eval. If we want to get this in before alpha, we'll need to at least QA related behaviors afterwards

**Comment by rundis** [comment]
The Inline result behavior is used by lot's of plugins (clojure, javascript, instabuster, clojure-light-refactor, python, haskell, etc etc). All of them I potentially suffering from the instarepl fix (or rather the dependent fix in eval.cljs in LT core).

From what I can figure out, it doesn't seem the clojure instarepl is actually using the behavior from eval, it rolls it's own. So I'm guessing(/hoping perhaps) that any potentially negative ramifications of this pull would be limited to bordeline cases where you switch from normal repl to instarepl and back, potentially with watches and what not. Allthough to me it seems a full clean of widgets happens when you do this switch anyway.

I 'd love to come up with a definitive convincing story, but I guess I'm not quite there yet.

I did test quite a bit both with and without the instarepl (and watches) for this fix, but of course no guarantees can be made until I actually fully understand the instarepl impl.

To summarize my ramblings:
- Fixing a generic behavior used by potentially lots of plugins should trump potentially breaking some edge cases in the clojure instarepl
- The fix only fixes an annoyance (simple workaround to clean inline results first), not a vital fix by any means so it shouldn't hold back an alpha-release unless we're talking hours or less in difference
- I'd be inclined to shoot from the hip and include it, but then again I'm biased :-)

**Comment by cldwalker** [comment]
@rundis Agreed keeping generic eval is more important than a one-off fix for instarepl. Any reason this PR doesn't just revert 2b0e9086927f5870966b9937efbf1c0e6ae47d95?

**Comment by rundis** [comment]
That would have been to easy :-)

**Comment by rundis** [comment]
maybe this line should stay ? https://github.com/LightTable/LightTable/pull/2008/files#diff-7bc0013fd8e35583a4681d44ccf83a91L485

**Comment by cldwalker** [comment]
I looked into this more and understand what happened. LightTable/Clojure@3017298 is the correct fix for instarepl stacking. Since instarepl/update-res uses inline objects it was thought that eval/inline-results, an inline behavior, also needed to be updated. However, this is false. If you search for the associated :editor.result trigger it is only found in clojure.cljs which is used by clj + cljs project, _not_ instarepl. I observed this to be true with print debugging.

> maybe this line should stay ? https://github.com/LightTable/LightTable/pull/2008/files#diff-7bc0013fd8e35583a4681d44ccf83a91L485

Given this commit was wrong in other ways, I'm ok with also reverting this until we observe an issue with 'Clear inline results'. It seems to work fine in clj, cljs and instarepl

@rundis Thanks for ringing the alarm and please merge! :+1:

**Comment by rundis** [comment]
Fixed #2004
